src/agentPreview.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { randomUUID } from 'node:crypto';
+import { SfError } from '@salesforce/core';
+import { Connection } from '@salesforce/core';
+import { MaybeMock } from './maybe-mock';
+
+type ApiStatus = {
+  status: 'UP' | 'DOWN';
+};
+
+type Href = { href: string };
+
+export type AgentPreviewError = {
+  status: number;
+  path: string;
+  requestId: string;
+  error: string;
+  message: string;
+  timestamp: number;
+};
+
+export type AgentPreviewMessageLinks = {
+  self: Href | null;
+  messages: Href | null;
+  session: Href | null;
+  end: Href | null;
+};
+
+export type MessageType =
+  | 'Inform'
+  | 'TextChunk'
+  | 'ProgressIndicator'
+  | 'Inquire'
+  | 'Confirm'
+  | 'Failure'
+  | 'Escalate'
+  | 'SessionEnded'
+  | 'EndOfTurn'
+  | 'Error';
+
+export type AgentPreviewMessage = {
+  type: MessageType;
+  id: string;
+  feedbackId: string;
+  planId: string;
+  isContentSafe: boolean;
+  message: string;
+  result: {
+    type: string;
+    property: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    value: any;
+  };
+  citedReferences: {
+    type: string;
+    value: string;
+  };
+};
+
+export type AgentPreviewStartResponse = {
+  sessionId: string;
+  _links: AgentPreviewMessageLinks;
+  messages: AgentPreviewMessage[];
+};
+
+export type AgentPreviewSendResponse = {
+  messages: AgentPreviewMessage[];
+  _links: AgentPreviewMessageLinks;
+};
+
+export type AgentPreviewEndResponse = {
+  messages: {
+    type: string;
+    id: string;
+    reason: string;
+    feedbackId: string;
+  };
+  _links: AgentPreviewMessageLinks;
+};
+
+type EndReason = 'UserRequest' | 'Transfer' | 'Expiration' | 'Error' | 'Other';
+
+export class AgentPreview {
+  private apiBase: string;
+  private instanceUrl: string;
+  private maybeMock: MaybeMock;
+
+  public constructor(connection: Connection) {
+    this.apiBase = 'https://api.salesforce.com/einstein/ai-agent/v1';
+    this.instanceUrl = connection.instanceUrl;
+    this.maybeMock = new MaybeMock(connection);
+  }
+
+  public async start(botId: string): Promise<AgentPreviewStartResponse> {
+    const url = `${this.apiBase}/agents/${botId}/sessions`;
+
+    const body = {
+      externalSessionKey: randomUUID(),
+      instanceConfig: {
+        endpoint: this.instanceUrl,
+      },
+      streamingCapabilities: {
+        chunkTypes: ['Text'],
+      },
+      bypassUser: true,
+    };
+
+    try {
+      return await this.maybeMock.request<AgentPreviewStartResponse>('POST', url, body);
+    } catch (err) {
+      const error = err as Error;
```

Comment:
`SfError.wrap(err)` handles this better to avoid the cast.  (same comment for the other catches)